notebook/notebookapp.py

```diff
@@ -155,7 +155,7 @@ def init_settings(self, ipython_app, kernel_manager, contents_manager,
             "template_path",
             ipython_app.template_file_path,
         )
-        if isinstance(_template_path, str):
+        if isinstance(_template_path, py3compat.string_types):
```

Comment:
Probably historical reasons. it does not make much differences. Also I suppose people could use many kind of iterables, but string are iterables, so "isiterable" wouldn't work.